src/components/lockable/lockable.cairo

```diff
@@ -1,3 +1,100 @@
+#[starknet::component]
+pub mod LockableComponent {
+    // *************************************************************************
+    //                              IMPORTS
+    // *************************************************************************
+    use starknet::storage::StoragePointerWriteAccess;
+    use starknet::storage::StoragePointerReadAccess;
+    use starknet::{ContractAddress, get_caller_address, get_block_timestamp};
+    use token_bound_accounts::components::account::account::AccountComponent;
+    use token_bound_accounts::interfaces::IAccount::{IAccount, IAccountDispatcherTrait};
+    use token_bound_accounts::components::account::account::AccountComponent::InternalImpl;
+    use token_bound_accounts::interfaces::ILockable::{
+        ILockable, ILockableDispatcher, ILockableDispatcherTrait
+    };
 
+    #[storage]
+    pub struct Storage {
+        lock_until: u64
+    }
+
+    #[event]
+    #[derive(Drop, starknet::Event)]
+    pub enum Event {
+        AccountLocked: AccountLocked
+    }
+
+    /// @notice Emitted when the account is locked
+    /// @param account tokenbound account who's lock function was triggered
+    /// @param locked_at timestamp at which the lock function was triggered
+    /// @param duration time duration for which the account remains locked
+    #[derive(Drop, starknet::Event)]
+    pub struct AccountLocked {
+        #[key]
+        pub account: ContractAddress,
+        pub locked_at: u64,
+        pub lock_until: u64,
+    }
+
+
+    // *************************************************************************
+    //                              ERRORS
+    // *************************************************************************
+    pub mod Errors {
+        pub const UNAUTHORIZED: felt252 = 'Account: unauthorized';
+        pub const NOT_OWNER: felt252 = 'Not Account Owner';
+        pub const EXCEEDS_MAX_LOCK_TIME: felt252 = 'Lock time exceeded';
+        pub const LOCKED_ACCOUNT: felt252 = 'Account Locked';
+    }
+
+
+    // storage that store the token_id and the lock_util perioed
+
+    // *************************************************************************
+    //                              EXTERNAL FUNCTIONS
+    // *************************************************************************
+    #[embeddable_as(LockableImpl)]
+    pub impl Lockable<
+        TContractState,
+        +HasComponent<TContractState>,
+        +Drop<TContractState>,
+        impl Account: AccountComponent::HasComponent<TContractState>
+    > of ILockable<ComponentState<TContractState>> {
+        fn lock(ref self: ComponentState<TContractState>, lock_until: u64) {
+            let current_timestamp = get_block_timestamp();
+
+            let account_comp = get_dep_component!(@self, Account);
+
+            // get the token owner
+            let owner = account_comp.owner();
+
+            assert(get_caller_address() == owner, Errors::NOT_OWNER);
```

Comment:
let's replace this logic that gets the owner and checks again the caller address with the `is_valid_signer` method on Account, since it does something similar but better for readability